src/components/DataTable/hoc/withRowLoadingPlaceholder.tsx

```diff
@@ -15,13 +15,15 @@ export const withRowLoadingPlaceholder = (
         const isLoaded = useDataTable(store => !!store.hasRowLoaded(props.index));
 
         const placeHolderRowStyle = useMemo(
-            () => [styles.placeHolderRow, props.style],
-            [props.style],
+            () => (isLoaded ? [styles.placeHolderRow, props.style] : undefined),
+            [props.style, isLoaded],
         );
 
-        if (!isLoaded) return <View style={placeHolderRowStyle} ref={ref} />;
-
-        return <Component {...props} ref={ref} />;
+        return (
+            <View style={placeHolderRowStyle} ref={ref}>
```

Comment:
Why is the ref passed to 2 places?
One will overwrite another

Reply:
Yeah, it's fine. This is a valid pattern, it allows you to maintain a single ref for both components. As long as `Component` handles the forwarded ref correctly, and any operations performed using the ref are appropriate for both components.